Yoast/wordpress-seo
# Make the dashboard widget display in the correct locale

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where the Dashboard Widget was not displayed in the correct language.

## Relevant technical choices:

Quoting our software architect:
> for any endpoints we create that include strings meant for UI we should use the user locale

This PR switches to the user locale before getting the Dashboard Widget statistics and labels.

## Test instructions
- first, check on trunk
- switch the user language (from Users > Your Profile) to a language other than English, for example: Italian
- go to the Dashboard and observe the UI is in Italian but various strings in the widget are still in English

<img width="656" alt="screen shot 2018-10-17 at 13 02 19" src="https://user-images.githubusercontent.com/1682452/47082553-41cb0780-d20e-11e8-995b-1f3b02229302.png">

- switch to this branch
- `cd` to your WordPress VVV install and ssh into VVV, then flush the transients: this is necessary because the statistics string are cached in a transient, it's a separate issue see #5521)
- in a typical VVV install, once you're ssh'd you need to run these two commands:
  - `cd /srv/www/wordpress-develop/public_html/`
  - `wp transient delete-all`
- In your WordPress Admin go to "Dashboard->Updates", scroll down and press the button to update all translations. If you later switch to a language you have not used before, click this button again. This downloads all the most recent translations.
- refresh the dashboard page
- verify the strings are now displayed in Italian:
- if somehow not all strings are translated run `wp transient delete-all` again. 

<img width="656" alt="screen shot 2018-10-03 at 14 00 54" src="https://user-images.githubusercontent.com/1682452/47082729-e4838600-d20e-11e8-9c8d-d747c9ab54d7.png">

Note: the string related to the `keyphrase` highlighted above has no available translation yet.

Question:
Should we restore to the previous locale after getting the statistics and labels? I guess no, because these are REST API calls but I'd appreciate any feedback. I'd guess doing something like the following is unnecessary:
```
if ( $locale_switched ) {
    restore_previous_locale();
}
```

Fixes #8674
